### ISSUE TITLE
HistogramSummary  not support fp16 ？

### ISSUE THREAD

**Issue by zkk0911** [tensorflow/tensorflow#14848]
There is a issue when I train my model with adm optimizer (tf.train.AdamOptimizer),  It seems HistogramSummary not support DT_HALF.   

2017-11-24 09:19:36.954820: W tensorflow/core/framework/op_kernel.cc:1192] Invalid argument: Infinity in summary histogram for: local4/BatchNorm/beta_1
         [[Node: local4/BatchNorm/beta_1 = HistogramSummary[T=DT_HALF, _device="/job:localhost/replica:0/task:0/device:CPU:0"](local4/BatchNorm/beta_1/tag, local4/BatchNorm/beta/read/_171)]]

But if i use the other optimizer such as GradientDescentOptimizer or RMSPropOptimizer， there is no the issue above.  I don't know why

**Comment by aselle** [comment]
Does it work fine if you use a 10th of the learning rate. It looks like you are hitting infinities in your optimization. That can happen if the optimization is at the edge of stability. Using reduced precision makes numerics less stable.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue? Please update the label and/or status accordingly.
